DigitalLearningSolutions.Web/Controllers/MyAccountController.cs

```diff
@@ -100,21 +100,28 @@ DlsSubApplication dlsSubApplication
             };
         }
 
-        private IActionResult EditDetailsPostSave(EditDetailsFormData formData, DlsSubApplication dlsSubApplication)
+        private IActionResult EditDetailsPostSave(
+            MyAccountEditDetailsFormData formData,
+            DlsSubApplication dlsSubApplication
+        )
         {
             var userAdminId = User.GetAdminId();
             var userDelegateId = User.GetCandidateId();
 
             if (userDelegateId.HasValue)
             {
-                ValidateJobGroup(formData);
-                ValidateCustomPrompts(formData);
+                if (!formData.JobGroupId.HasValue)
+                {
+                    ModelState.AddModelError(nameof(EditDetailsFormData.JobGroupId), "Select a job group");

```

Comment:
You can achieve this with a custom validator in the View Model (see SummaryViewModel for an example). I think it's better to do validation in the View Model where possible than in the controller, but I'm open to being persuaded otherwise.

Reply:
So this behaves a bit oddly. I have to have the IValidatableObject at the level above the EditDetailsFormData, the type on the method parameter, otherwise it doesn't work. I can have the Validate method ishared in EditDetailsFormData though.
This also means I've had to remove some test assertions since they don't actually validate.